dotCMS/src/main/java/com/dotcms/saml/DotSamlFactory.java

```diff
@@ -0,0 +1,102 @@
+package com.dotcms.saml;
+
+import com.dotcms.saml.service.external.IdentityProviderConfigurationFactory;
+import com.dotcms.saml.service.external.MessageObserver;
+import com.dotcms.saml.service.external.SamlAuthenticationService;
+import com.dotcms.saml.service.external.SamlConfigurationService;
+import com.dotcms.saml.service.external.SamlException;
+import com.dotcms.saml.service.external.SamlServiceBuilder;
+import com.dotcms.security.apps.AppDescriptor;
+import com.dotcms.security.apps.AppsAPI;
+import com.dotmarketing.business.APILocator;
+import com.liferay.portal.model.User;
+import io.vavr.control.Try;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class DotSamlFactory {
+
+    public static final String SAML_APP_CONFIG_KEY = "dotsaml-config";
+
+    private final Map<Class, Object> instanceMap        = new ConcurrentHashMap<>();
+    private final SamlServiceBuilder samlServiceBuilder = null; // todo: get this from osgi reference
+    private final MessageObserver    messageObserver    = new DotLoggerMessageObserver();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Field messageObserver has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)